modules/appeals_api/spec/services/appeals_api/remove_pii_spec.rb

```diff
@@ -3,8 +3,158 @@
 require 'rails_helper'
 require AppealsApi::Engine.root.join('spec', 'spec_helper.rb')
 
+def update_appeal_status(appeal, status, code: nil, detail: nil)
+  # At the time of writing, the `update_status` method for each appeal model involves kicking off a sidekiq job to
+  # create a matching StatusUpdate record. This is unwieldy in tests, so this method approximates the
+  # `appeal.update_status!` method without involving sidekiq.
+  appeal.update!(status:, code:, detail:)
+
+  AppealsApi::StatusUpdate.create!(
+    from: appeal.status,
+    to: status,
+    status_update_time: Time.zone.now,
+    statusable_id: appeal.id,
+    statusable_type: appeal.class.name,
+    code:,
+    detail:
+  )
+
+  appeal
+end
+
+shared_examples 'removes expired PII' do
+  let(:now) { Time.zone.now }
+  let(:code) { 'DOC202' }
+  let(:detail) { "Upstream status: #{VBADocuments::UploadSubmission::ERROR_UNIDENTIFIED_MAIL}" }
+  let(:misc_appeal_types) do
+    %i[minimal_notice_of_disagreement extra_notice_of_disagreement_v2 extra_notice_of_disagreement_v0
+       higher_level_review_v1 extra_higher_level_review_v2 minimal_higher_level_review_v0
+       extra_supplemental_claim minimal_supplemental_claim_v0]
+  end
+
+  def create_appeals = [FactoryBot.create(v0_factory), FactoryBot.create(v2_factory)]
+
+  def create_misc_appeals = create_appeals + misc_appeal_types.map { |f| FactoryBot.create(f) }
+
+  let!(:unexpired_appeals) do # These should all be ignored and remain unchanged
```

Comment:
Examples in this section are all intended to be close to the removal criteria but not quite matching - these should be ignored.